### DISCUSSION TITLE
How to use unique identifier in key prop when mapping splitAtom ?

### DISCUSSION THREAD

**Discussion by smontlouis** [pmndrs/jotai#1602] [Q&A] [answered]
I'm using splitAtom to handle tabs reordering, adding and removing.

```tsx 
const tabsAtom = atom<TabItem[]>([
  { id, type, data, ....},
  { id, type, data, ....},
  { id, type, data, ....},
])

const tabsAtomsAtom = splitAtom(tabsAtom)

// in component
const [tabsAtoms, dispatch] = useAtom(tabsAtomsAtom)

<ScrollView>
  {tabsAtoms.map(
    (tabAtom, i) =>
        <TabPreview
          key={i}
          index={i}
          tabAtom={tabAtom}
          tapGestureRef={tapGestureRefs[i]}
          simultaneousHandlers={scrollViewRef}
          onPress={onItemPress}
          onDelete={onDeleteItem}
        />
  )}
</ScrollView>
```

I'm using reanimated to animate entering / exiting animations, but it fail to work properly.
I believe this is due to the fact I'm using `i` as key.

Is there any way to use an atom unique identifier ?

**Comment by dai-shi** [comment, marked as answer]
That's what `keyExtractor` is for.

```jsx
const tabsAtomsAtom = splitAtom(tabsAtom, item => item.id)

  {tabsAtoms.map(
    (tabAtom, i) =>
        <TabPreview
          key={`${tabAtom}`}
```

**Reply by smontlouis** [reply]
Awesome ! Should have looked in the splitAtom signature, because it's missing in the doc :).

[Unfortunately it does nothing](https://codesandbox.io/s/sweet-pascal-k7dcoi?file=/src/App.js)

**Reply by dai-shi** [reply]
> it's missing in the doc :).

Yeah, we should improve it.

> Unfortunately it does nothing

Maybe, it's different from `key` issue. Can you make it to represent your issue?

**Reply by smontlouis** [reply]
I need "xxx" and "ccc" as key for my component, but I have "atom15, atom16". 

What is the purpose of the keyExtractor then ?


edit: nvm I understand the purpose of keyExtractor, and it's working !
I thought there would some visual change also.

Thank you !

**Reply by dai-shi** [reply]
Yeah, it will re-use the atom object based on "keys".

If you are interested, I would appreciate very much if you could improve the doc as you seem to understand how things work in the specific use case.

**Reply by smontlouis** [reply]
Not a native speaker, but will try my best ;)

**Reply by saoudi-h** [reply]
Great! That helped, but unfortunately, this is not in the doc. I might be wrong

**Reply by dai-shi** [reply]
Will you send a PR please?
